### PR TITLE
Migrate from Ollama `generate` api to `chat` api

### LINKED ISSUE #15: Use the Ollama chat API instead of the generate API
It currently uses the `/api/generate` endpoint to generate one-off responses. The `/api/chat` endpoint fits our use case better, and is better for integrating history/context

### PR DESCRIPTION
**Pull request by eamonburns**

### CODE REVIEW

File: app/bot/llm.py [hunk 1]
```diff
@@ -28,10 +28,13 @@ async def generate_response(
...
-                    "prompt": message.content,
-                    "system": system_prompt,
-                    "context": [],
                     "stream": False,
+                    "messages": [
+                        {"role": "system",
+                        "content": system_prompt},
+                        {"role": "user",
+                        "content": message.content},
+                    ],
```
Comment (joshbuker) [inline app/bot/llm.py:37]: Still doesn't have history over time, but this does get you closer to migrating and should be functionally equivalent.

Comment (joshbuker) [inline app/bot/llm.py:37]: Trailing comma, but I suspect Python is like most languages and doesn't care

**Review by joshbuker — APPROVED** [review]
LGTM

File: app/bot/llm.py (hunk 1 above)
Comment (eamonburns) [reply app/bot/llm.py:37]: Yes, this is a quick change to just migrate to the chat API. History will come later

Comment (eamonburns) [reply app/bot/llm.py:37]: Nope, it doesn't care. I also think they look better (it was on purpose)

Comment (joshbuker) [reply app/bot/llm.py:37]: Fair enough, lol

**Pull request merged**